indico_toolkit/metrics/compare_ground_truth.py

```diff
@@ -0,0 +1,119 @@
+            }
+        else:
+            raise ToolkitInputError("Invalid span type entered.")
+
+    def set_overall_metrics(self) -> None:
+        if self.all_label_metrics is None:
+            self.set_all_label_metrics()
+
+        metrics_types = [
+            "true_positives",
+            "false_positives",
+            "false_negatives",
+        ]
+
+        self.overall_metrics = {
+            "true_positives": 0,
+            "false_positives": 0,
+            "false_negatives": 0,
+        }
+
+        for metric in metrics_types:
+            total_amt = 0
+            for label in self.all_label_metrics:
+                total_amt += self.all_label_metrics[label][metric]
+                self.overall_metrics[metric] = total_amt
+
+        self.overall_metrics["precision"] = self._get_precision(
+            self.overall_metrics["true_positives"],
+            self.overall_metrics["false_positives"],
+        )
+        self.overall_metrics["recall"] = self._get_recall(
+            self.overall_metrics["true_positives"],
+            self.overall_metrics["false_negatives"],
+        )
+
+    def _get_precision(self, true_p: int, false_p: int) -> float:
+        try:
+            precision = true_p / (true_p + false_p)
+        except ZeroDivisionError:
+            precision = 0
+        return precision
+
+    def _get_recall(self, true_p: int, false_n: int) -> float:
+        try:
+            recall = true_p / (true_p + false_n)
+        except ZeroDivisionError:
+            recall = 0
+        return recall
+
+    # TODO add in other span type functions (), ultimately move them to toolkit, within association
+    def _sequences_exact(self, x: dict, y: dict) -> bool:
+        """
+        Boolean return value indicates whether or not seqs are exact
+        """
+        return x["start"] == y["start"] and x["end"] == y["end"]
+
+    def _get_base_metrics(self, label: str, span_type: str) -> dict:
+        """
+        With the current overlap span type calculation, if 2 separate predictions
+        each overlap with a single ground truth, each pred is counted as a true
+        positive. (i.e. There isn't a break out of the loop once a TP is found.)
+        """
+        true_pos = 0
+        false_neg = 0
+        false_pos = 0
+        span_types = {"overlap": sequences_overlap, "exact": self._sequences_exact}
+        span_type_func = span_types[span_type]
+
+        if not label in self.preds_by_label:
+            false_neg = len(self.gt_by_label[label])
+        elif not label in self.gt_by_label:
+            false_pos = len(self.preds_by_label[label])
+        else:
+            for model_pred in self.preds_by_label[label]:
+                match_found = False
+                for gt_pred in self.gt_by_label[label]:
+                    if span_type_func(model_pred, gt_pred):
+                        true_pos += 1
+                        match_found = True
```

Comment:
we could allow a choice on the "multiple true positives" per ground truth prediction by adding a conditional in here... something to think about